src/transformers/models/llama/modeling_llama.py

```diff
@@ -138,16 +140,11 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None, s
         self.scaling_factor = scaling_factor
         super().__init__(dim, max_position_embeddings, base, device)
 
-    def _set_cos_sin_cache(self, seq_len, device, dtype):
-        self.max_seq_len_cached = seq_len
-        t = torch.arange(self.max_seq_len_cached, device=device, dtype=torch.int64).type_as(self.inv_freq)
-        t = t / self.scaling_factor
-
-        freqs = torch.outer(t, self.inv_freq)
-        # Different from paper, but it uses a different permutation in order to obtain the same calculation
-        emb = torch.cat((freqs, freqs), dim=-1)
-        self.register_buffer("cos_cached", emb.cos().to(dtype), persistent=False)
-        self.register_buffer("sin_cached", emb.sin().to(dtype), persistent=False)
+    def forward(self, x, position_ids, seq_len=None):
```

Comment:
I am alright with this but it is breaking for any libs that rely on sin cached and cos cached. Same for the static cache PR! 
Let's just add a mention that it will be removed next release and still compute cos and sin!

Reply:
This is the cool part -- it calls super's forward, which in turn caches sin/cos (see [here](https://github.com/huggingface/transformers/blob/1a77f07f6556b1482bd5e5f8399aa528727d1b47/src/transformers/models/llama/modeling_llama.py#L105)). BC is preserved 🙌

Reply:
Yes but we need a warning to deprecate ! 
Follow up is fine

Reply:
I'm not sure I follow -- the warning is [here](https://github.com/huggingface/transformers/blob/58245ba6fba739a03388f52b0773d2fc965701e3/src/transformers/models/llama/modeling_llama.py#L105). Or were you thinking of some other warning?